python/ray/serve/_private/application_state.py

```diff
@@ -424,17 +425,23 @@ def _determine_app_status(self) -> Tuple[ApplicationStatus, str]:
             return ApplicationStatus.DELETING, ""
 
         num_healthy_deployments = 0
+        num_autoscaling_deployments = 0
+        num_updating_deployments = 0
         unhealthy_deployment_names = []
 
         for deployment_status in self.get_deployments_statuses():
             if deployment_status.status == DeploymentStatus.UNHEALTHY:
                 unhealthy_deployment_names.append(deployment_status.name)
-            if deployment_status.status == DeploymentStatus.HEALTHY:
+            elif deployment_status.status == DeploymentStatus.HEALTHY:
                 num_healthy_deployments += 1
+            elif (
+                deployment_status.status_trigger == DeploymentStatusTrigger.AUTOSCALING
+            ):
+                num_autoscaling_deployments += 1
+            else:
```

Comment:
This conditioning is brittle -- we add a new state and this will silently break.

Instead, do it explicit:

```
if HEALTHY:
  ...
elif UNHEALTHY:
  ...
elif UPDATING:
  ...
else:
  raise RuntimeError
```

Reply:
Updated!